controller/controller_server/csi_controller_server.py

```diff
@@ -283,54 +260,24 @@ def ControllerPublishVolume(self, request, context):
-
-        except array_errors.PermissionDeniedError as ex:
-            context.set_code(grpc.StatusCode.PERMISSION_DENIED)
-            context.set_details(ex.message)
-            return csi_pb2.ControllerPublishVolumeResponse()
-
+            return handle_exception(ex, context, grpc.StatusCode.FAILED_PRECONDITION,
+                                    csi_pb2.ControllerPublishVolumeResponse)
         except (array_errors.LunAlreadyInUseError, array_errors.NoAvailableLunError) as ex:
-            logger.exception(ex)
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.RESOURCE_EXHAUSTED)
-            return csi_pb2.ControllerPublishVolumeResponse()
-
-        except (array_errors.HostNotFoundError, array_errors.ObjectNotFoundError,
-                array_errors.NoIscsiTargetsFoundError, ObjectIdError) as ex:
-            logger.exception(ex)
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-            return csi_pb2.ControllerPublishVolumeResponse()
-
-        except (ValidationException, array_errors.IllegalObjectID,
-                array_errors.UnsupportedConnectivityTypeError) as ex:
-            logger.exception(ex)
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
-            return csi_pb2.ControllerPublishVolumeResponse()
-
-        except Exception as ex:
-            logger.debug("an internal exception occurred")
-            logger.exception(ex)
-            context.set_code(grpc.StatusCode.INTERNAL)
-            context.set_details('an internal exception occurred : {}'.format(ex))
-            return csi_pb2.ControllerPublishVolumeResponse()
-
+            return handle_exception(ex, context, grpc.StatusCode.RESOURCE_EXHAUSTED,
+                                    csi_pb2.ControllerPublishVolumeResponse)
+        except (array_errors.NoIscsiTargetsFoundError, ObjectIdError) as ex:
+            return handle_exception(ex, context, grpc.StatusCode.NOT_FOUND, csi_pb2.ControllerPublishVolumeResponse)
+        except array_errors.UnsupportedConnectivityTypeError as ex:
+            return handle_exception(ex, context, grpc.StatusCode.INVALID_ARGUMENT,
+                                    csi_pb2.ControllerPublishVolumeResponse)
+
+    @handle_common_exceptions(csi_pb2.ControllerUnpublishVolumeResponse)
     def ControllerUnpublishVolume(self, request, context):
         set_current_thread_name(request.volume_id)
         logger.info("ControllerUnpublishVolume")
+        utils.validate_unpublish_volume_request(request)
+
         try:
-            try:
-                utils.validate_unpublish_volume_request(request)
-            except ValidationException as ex:
-                logger.exception(ex)
-                context.set_details(ex.message)
-                context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
-                return csi_pb2.ControllerUnpublishVolumeResponse()
 
```

Comment:
I think this empty line should be removed as well

Reply:
Done